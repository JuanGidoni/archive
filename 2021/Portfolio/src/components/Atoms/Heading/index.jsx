import './index.css';

const Heading = ({ type, text, className }) => {
 const h1 = <h1 className={className}>{text}</h1>;
 const h2 = <h2 className={className}>{text}</h2>;
 const h3 = <h3 className={className}>{text}</h3>;

 if (type === "h1") return h1;
 if (type === "h2") return h2;

 return h3;
};

export default Heading;
