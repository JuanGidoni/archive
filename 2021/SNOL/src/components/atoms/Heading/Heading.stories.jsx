import React from "react";
import Heading from "./index";

export default {
  title: "Heading",
  argTypes: {
    variant: {
      options: ["Light Mode", "Dark Mode"],
      control: { type: "radio" },
    },
    type: {
      options: ["h1", "h2", "h3"],
      control: { type: "radio" },
    }
  },
  args: {
    className: "text-3xl text-indigo-600 font-medium w-full text-left",
    text: "Welcome",
  },
  component: Heading,
};

export const HeadingExample = (args) => <Heading {...args} />;
