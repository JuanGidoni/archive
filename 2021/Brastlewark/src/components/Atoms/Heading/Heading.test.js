import React from 'react';
import renderer from 'react-test-renderer';
import Heading from './Heading';

describe('<Heading />', () => {
 it('renders correctly h1 and match snapshot', () => {
  const tree = renderer
   .create(<Heading type="h1" text="Example h1" className="h1" />)
   .toJSON();
  expect(tree).toMatchSnapshot();
 });
 it('renders correctly h2 and match snapshot', () => {
  const tree = renderer
   .create(<Heading type="h2" text="Example h2" className="h2" />)
   .toJSON();
  expect(tree).toMatchSnapshot();
 });
 it('renders correctly h3 and match snapshot', () => {
  const tree = renderer
   .create(<Heading text="Example h3" className="h3" />)
   .toJSON();
  expect(tree).toMatchSnapshot();
 });
});
