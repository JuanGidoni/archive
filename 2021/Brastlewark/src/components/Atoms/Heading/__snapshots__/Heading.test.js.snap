// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders correctly h1 and match snapshot 1`] = `
<h1
  className="h1"
>
  Example h1
</h1>
`;

exports[`<Heading /> renders correctly h2 and match snapshot 1`] = `
<h2
  className="h2"
>
  Example h2
</h2>
`;

exports[`<Heading /> renders correctly h3 and match snapshot 1`] = `
<h3
  className="h3"
>
  Example h3
</h3>
`;
